packages/ingest/src/HttpProviderClient.js

```diff
@@ -62,14 +64,35 @@ class HttpProviderClient {
 
     if (this.username) this.gotOptions.username = this.username;
     if (this.password) this.gotOptions.password = this.password;
+
+    const RedirectHandler = {
+      // Need to use named function and not fat arrow
+      // expression so that we can use the bound value
+      // of "this"
```

Comment:
You know generally I'm not a fan of expository comments as we should be writing code that's relatively self-documenting via explicit idioms and maintainable/testable-over-clever structures absent performance concerns, etc. 

In this case the reason for using this isn't tightly coupled to the code in a casual read-through, so the comment is *probably* warranted. 

The alternate is we have a test that more-or-less explicitly covers this and contains the/a comment/commit comment so if an update occurs it fails and it doesn't eat an hour of engineering time to recall 'why'.

Reply:
I guess I'm not wedded to having a comment, but it is a departure from most of our code to not use fat arrow functions (`() =>`). However, there's a very specific reason in that we need to use a named function so that the lexical value of `this` is preserved

In terms of unit test vs a comment, the unit test to "prove" that it is working is really any one of the unit tests I've added to prove that redirects are properly handled. So I don't think another specific unit test is needed, but I could be wrong

If people feel strongly I should remove the comment, let me know